mbuild/conversion.py

```diff
@@ -0,0 +1,1386 @@
+    smiles: bool, optional, default=False
+        Use Open Babel to parse filename as a SMILES string
+        or file containing a SMILES string.
+    infer_hierarchy : bool, optional, default=True
+        If True, infer hierarchy from chains and residues
+    **kwargs : keyword arguments
+        Key word arguments passed to mdTraj for loading.
+
+    Returns
+    -------
+    compound : mb.Compound
+
+    """
+    # First check if we are loading from an object)
+    if not isinstance(filename_or_object, str):
+        return load_object(
+                        object=filename_or_object,
+                        compound=compound,
+                        coords_only=coords_only,
+                        rigid=rigid,
+                        infer_hierarchy=infer_hierarchy,
+                        **kwargs)
+    # Second check if we are loading SMILES strings
+    elif smiles:
+        return load_smiles(
+                        smiles_or_filename=filename_or_object,
+                        compound=compound,
+                        infer_hierarchy=infer_hierarchy)
+    # Last, if none of the above, load from file
+    else:
+        return load_file(
+                        filename=filename_or_object,
+                        relative_to_module=relative_to_module,
+                        compound=compound,
+                        coords_only=coords_only,
+                        rigid=rigid,
+                        backend=backend,
+                        infer_hierarchy=infer_hierarchy,
+                        **kwargs)
+
+def load_object(object, compound=None, coords_only=False,
+            rigid=False, infer_hierarchy=True, **kwargs):
+    """ Helper function to load an object into a mb.Compound
+
+    Functions to load on-disk object to mb.Compound, supporting conversion
+    from mb.Compound, pmd.Structure, md.Trajectory, and pybel.Molecule. If the
+    compound flag is supplied, this will add the object as a subcompound of that
+    compound.
+
+    Parameters
+    ----------
+    object :
+        Object to be converted.
+    compound : mb.Compound, optional, default=None
+        The host mbuild Compound
+    coords_only : bool, optional, default=False
+        Only load the coordinates into existing compound.
+    rigid : bool, optional, default=False
+        Treat the compound as a rigid body
+    infer_hierarchy : bool, optional, default=True
+        If True, infer hiereachy from chains and residues
+    **kwargs : keyword arguments
+        Key wordd arguments passed to mdTraj for loading
+
+    Returns
+    -------
+    mb.Compound
+
+    """
+    # Create type_dict type -> loading method
+    type_dict = {
+        mb.Compound:from_mbuild,
+        pmd.Structure:from_parmed,
+        md.Trajectory:from_trajectory
+                }
+    try:
+        pybel = import_('pybel')
+        type_dict.update({pybel.Molecule:from_pybel})
+    except ImportError:
+        pass
```

Comment:
```suggestion
    if has_openbabel:
        pybel = import_('pybel')
        type_dict.update({pybel.Molecule:from_pybel})
```

This way you don't get an error message every time you try to load something if open babel isn't installed. You might have to import `has_openbabel` from `mbuild.utils.io`